### PR TITLE
Email unsubscribe request dashboard

### PR DESCRIPTION
**Pull request by rparke**

### CODE REVIEW

File: app/dao/unsubscribe_request_dao.py [hunk 1]
```diff
@@ -10,3 +10,18 @@ def create_unsubscribe_request_dao(unsubscribe_data):
 
 def get_unsubscribe_request_by_notification_id_dao(notification_id):
     return UnsubscribeRequest.query.filter_by(notification_id=notification_id).one()
+
+
+def get_count_of_unprocessed_requests(notification_id):
```
Comment (rparke) [inline app/dao/unsubscribe_request_dao.py:15]: rename to be suffixed with dao

File: app/dao/unsubscribe_request_dao.py [hunk 2]
```diff
@@ -10,3 +10,18 @@ def create_unsubscribe_request_dao(unsubscribe_data):
...
+    return (
+        db.session.query(
+            UnsubscribeRequest.id.label("unsubscribe_request_id"),
+        )
+        .select_from(UnsubscribeRequest)
+        .join(
```
Comment (rparke) [inline app/dao/unsubscribe_request_dao.py:21]: join may not be needed in the first instance, check.

Filter on service_id in unsubscribe request table

File: app/unsubscribe_requests/rest.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+from datetime import datetime
+from app.dao.unsubscribe_request_dao import get_count_of_unprocessed_requests
+
+
+
+unsubscrube_requests_blueprint = Blueprint("unsubscribe_requests", __name__)
+
+@unsubscrube_requests_blueprint.route("/unsubscribe/<uuid:notification_id>/summary", methods=["GET"])
+def unsubscribe_requests_summary(notification_id):
+    pending_unsubscribe_requests = 250
```
Comment (rparke) [inline app/unsubscribe_requests/rest.py:11]: remove, just there for testing

File: app/unsubscribe_requests/rest.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    pending_unsubscribe_requests = get_count_of_unprocessed_requests(notification_id)
+    datetime_of_latest_unsubscribe_request = datetime.now()
+    return jsonify(pending_unsubscribe_requests=pending_unsubscribe_requests, datetime_of_latest_unsubscribe_request=datetime_of_latest_unsubscribe_request), 200
```
Comment (quis) [inline app/unsubscribe_requests/rest.py:14]: `count_of_pending_unsubscribe_requests` might be better (consistent with `datetime_of_latest_unsubscribe_request`):
```suggestion
    return jsonify(count_of_pending_unsubscribe_requests=pending_unsubscribe_requests, datetime_of_latest_unsubscribe_request=datetime_of_latest_unsubscribe_request), 200
```

**Comment by BlessedDev** [comment]
I have fleshed out `get_unsubscribe_requests_statistics_dao` and added unit tests.

**Comment by BlessedDev** [comment]
> Thanks for fleshing it out @BlessedDev, tests look good.
> 
> There’s some detail which got missed from the story but it’s in a Slack thread here (sorry): https://gds.slack.com/archives/C03S5QFTWSD/p1719835523440029
> 
> Here’s what I think the endpoint should return:
> 
> 1. If there are no unsubscribe requests in the database for a service:
>    `{}`
> 2. If there are some, but not in the last 7 days:
>    `{'count_of_pending_unsubscribe_requests': 0, 'datetime_of_latest_unsubscribe_request': '2001-01-01'}`
> 3. If there are 321 in the last 7 days and 456 in total:
>    `{'count_of_pending_unsubscribe_requests': 321, 'datetime_of_latest_unsubscribe_request': '2023-07-18'}`
> 
> Might need to rename the field from `…pending…` since that’s not accurate any more – the count should include requests in any state.

Ok, thanks for the link to the discussion, I missed the part about restricting it to 7 days. `count_of_pending_unsubscribe_requests` is confusing so I shall rename it as suggested.

**Review by quis — APPROVED** [review]

**Pull request merged**